File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/playback/file/volume/specs/GivenVolumeLevellingIsNotEnabled/WhenGettingTheMaxVolume.java
package com.lasthopesoftware.bluewater.client.playback.file.volume.specs.GivenVolumeLevellingIsNotEnabled;


import com.lasthopesoftware.bluewater.client.connection.IConnectionProvider;
import com.lasthopesoftware.bluewater.client.library.items.media.files.ServiceFile;
import com.lasthopesoftware.bluewater.client.library.items.media.files.properties.CachedFilePropertiesProvider;
import com.lasthopesoftware.bluewater.client.library.items.media.files.properties.IFilePropertiesProvider;
import com.lasthopesoftware.bluewater.client.library.items.media.files.properties.repository.IFilePropertiesContainerRepository;
import com.lasthopesoftware.bluewater.client.playback.file.volume.MaxFileVolumeProvider;
import com.lasthopesoftware.bluewater.settings.volumeleveling.IVolumeLevelSettings;
import org.junit.BeforeClass;
import org.junit.Test;

import java.util.concurrent.CountDownLatch;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.mock;

public class WhenGettingTheMaxVolume {

	private static float returnedVolume;

	@BeforeClass
	public static void before() throws InterruptedException {
		final CachedFilePropertiesProvider cachedFilePropertiesProvider =
			new CachedFilePropertiesProvider(
				mock(IConnectionProvider.class),
				mock(IFilePropertiesContainerRepository.class),
				mock(IFilePropertiesProvider.class));

		final IVolumeLevelSettings volumeLevelSettings = mock(IVolumeLevelSettings.class);

		final MaxFileVolumeProvider maxFileVolumeProvider =
			new MaxFileVolumeProvider(volumeLevelSettings, cachedFilePropertiesProvider);

		final CountDownLatch countDownLatch = new CountDownLatch(1);
		maxFileVolumeProvider
			.promiseMaxFileVolume(new ServiceFile(1))
			.then(volume -> {
				returnedVolume = volume;
				countDownLatch.countDown();
				return null;
			});

		countDownLatch.await();
	}

	@Test
	public void thenTheReturnedVolumeIsOne() {
		assertThat(returnedVolume).isEqualTo(1);
	}
}

File: projectBlueWater/src/test/java/com/lasthopesoftware/bluewater/client/playback/file/volume/GivenVolumeLevellingIsNotEnabled/WhenGettingTheMaxVolume.kt
package com.lasthopesoftware.bluewater.client.playback.file.volume.GivenVolumeLevellingIsNotEnabled

import com.lasthopesoftware.bluewater.client.browsing.files.ServiceFile
import com.lasthopesoftware.bluewater.client.browsing.library.repository.LibraryId
import com.lasthopesoftware.bluewater.client.playback.file.volume.MaxFileVolumeProvider
import com.lasthopesoftware.bluewater.shared.promises.extensions.toExpiringFuture
import com.namehillsoftware.handoff.promises.Promise
import io.mockk.every
import io.mockk.mockk
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class WhenGettingTheMaxVolume {
	private val returnedVolume by lazy {
		val maxFileVolumeProvider = MaxFileVolumeProvider(
			mockk {
				every { promiseIsVolumeLevellingEnabled() } returns Promise(false)
				every { promiseIsPeakLevelNormalizeEnabled() } returns Promise(true)
			},
			mockk()
		)

		maxFileVolumeProvider
			.promiseMaxFileVolume(LibraryId(244), ServiceFile("1"))
			.toExpiringFuture()
			.get()
	}

	@Test
	fun thenTheReturnedVolumeIsOne() {
		assertThat(returnedVolume).isEqualTo(1f)
	}
}
